src/ChainSafe.Gaming.WalletConnect/WalletConnectProvider.cs

```diff
@@ -233,6 +242,64 @@ public override async Task<string> Connect()
             }
         }
 
+        private async Task CheckAndSwitchNetwork()
+        {
+            var chainId = GetChainId();
+            if (chainId != $"{ChainModel.EvmNamespace}:{chainConfig.ChainId}")
```

Comment:
I think we should recursively check if the chain id is correct till it's correct here.

Reply:
You're right.

Reply:
Actually, no 😁  we are prompting them to switch, if they reject, we don't care that much because _every_ transaction that will happen in the future, will prompt them to switch to the corect chain due to the UpdateSessionChainId function being called right after it, which essentially sets the correct chain.

Reply:
aha, good then